app/src/test/kotlin/edu/stanford/bdh/engagehf/contact/ui/ContactScreenViewModelTest.kt

```diff
@@ -0,0 +1,87 @@
+package edu.stanford.bdh.engagehf.contact.ui
+
+import com.google.common.truth.Truth.assertThat
+import edu.stanford.bdh.engagehf.contact.data.EngageContactRepository
+import edu.stanford.spezi.core.navigation.NavigationEvent
+import edu.stanford.spezi.core.navigation.Navigator
+import edu.stanford.spezi.core.testing.CoroutineTestRule
+import edu.stanford.spezi.modules.contact.model.Contact
+import io.mockk.coEvery
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class ContactScreenViewModelTest {
+
+    @get:Rule
+    val coroutineTestRule = CoroutineTestRule()
+
+    private val engageContactRepository: EngageContactRepository = mockk(relaxed = true)
+    private val navigator: Navigator = mockk(relaxed = true)
+
+    private lateinit var viewModel: ContactScreenViewModel
+
+    @Before
+    fun setup() {
+        coEvery { engageContactRepository.getContact() } returns Result.success(mockk(relaxed = true))
+        viewModel = ContactScreenViewModel(
+            engageContactRepository = engageContactRepository,
+            navigator = navigator,
+        )
+    }
+
+    @Test
+    fun `loadContact should update state on success`() = runTest {
+        // Given
+        val contact = mockk<Contact>()
+        coEvery { engageContactRepository.getContact() } returns Result.success(contact)
+
+        // When
+        viewModel = ContactScreenViewModel(
```

Comment:
you can declare the view model lazily so that you can postpone its creation after you stubbed the repository. No need for init in setup method and each test then

```kotlin
private val viewModel by lazy {
    ContactScreenViewModel(
        engageContactRepository = engageContactRepository,
        navigator = navigator,
    )
}
```